controllers/controller.go

```diff
@@ -349,6 +353,12 @@ func (r *EtcdadmClusterReconciler) reconcile(ctx context.Context, etcdCluster *e
 		log.Info("Scaling down etcd cluster", "Desired", desiredReplicas, "Existing", numCurrentMachines)
 		// The last parameter corresponds to Machines that need to be rolled out, eg during upgrade, should always be empty here.
 		return r.scaleDownEtcdCluster(ctx, etcdCluster, cluster, ep, collections.Machines{})
+	case numCurrentMachines == desiredReplicas && etcdCluster.Status.Ready:
```

Comment:
There's an issue with the etcd upgrade in progress annotation not getting removed properly when only scaling an etcd cluster. 

We currently only remove this validation when etcd requires rollout and the rollout finishes. We need another case to handle removing this validation when scaling is complete so that kcp will roll out new nodes with the updated endpoints. Let's talk about an alternative source of truth for this.

Reply:
`case numCurrentMachines == desiredReplicas && conditions.IsTrue(etcdCluster, etcdv1.EtcdMachinesReadyCondition):`

maybe we should check the `etcdMachinesReadyCondition` instead? this is the aggregate of all the conditions in the etcd cluster and should be a source of truth. Let me know if you have any other ideas

Reply:
can we just use the info that drives `etcdv1.EtcdMachinesReadyCondition`? like whatever data we are using to set that to true of false

Reply:
We set this condition based on the aggregate of other conditions:

`	// This aggregates the state of all machines
	conditions.SetAggregate(etcdCluster, etcdv1.EtcdMachinesReadyCondition, ownedMachines.ConditionGetters(), conditions.AddSourceRef(), conditions.WithStepCounterIf(false))
`